code/modules/mapping/space_management/traits.dm

```diff
@@ -49,6 +49,15 @@
 				. += S.z_value
 				break
 
+/// Get a list of all z which have all of the specified traits
+/datum/controller/subsystem/mapping/proc/levels_by_all_traits(z, list/traits)
+	. = list()
+	var/list/_z_list = z_list
+	for(var/A in _z_list)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#avoid-unnecessary-type-checks-and-obscuring-nulls-in-lists
```suggestion
	for(var/datum/space_level/S as anything in _z_list)
```